### PR TITLE
 feat(pci.savings-plan): active plans consumption

### PR DESCRIPTION
**Pull request by Eric-ciccotti**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master` / `release/**` / `develop` 
| Bug fix?         | yes/no
| New feature?     | yes/no
| Breaking change? | yes/no
| Tickets          | Fix #... 
| License          | BSD 3-Clause

- [ ] Try to keep pull requests small so they can be easily reviewed.
- [ ] Commits are signed-off
- [ ] Only FR translations have been updated
- [ ] Branch is up-to-date with target branch
- [ ] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [ ] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/apps/pci-savings-plan/src/components/Dashboard/Kpis/Kpis.tsx [hunk 1]
```diff
@@ -49,53 +53,60 @@ const Kpi = ({
...
       </OdsText>
       <span className="text-[20px] font-bold mb-0 text-[#0050D7]">{value}</span>
+      {valueSavedWithoutAmount && (
+        <div className="flex items-center">
+          <span className="text-[15px] mb-0 text-[#666ca1] mr-2">
+            {t('dashboard_kpis_saved_strikethrough')}
+          </span>
+          <span className="line-through text-[15px] mb-0 text-[#666ca1]">
+            {valueSavedWithoutAmount}
+          </span>
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-savings-plan/src/components/Dashboard/Kpis/Kpis.tsx:63]: Price should be integrated inside the translation to be able to move it where it belongs in the sentence

I think we should also consider putting the other price inside to make a full sentence

File: packages/manager/apps/pci-savings-plan/src/components/Dashboard/Kpis/Kpis.tsx [hunk 2]
```diff
@@ -49,53 +53,60 @@ const Kpi = ({
...
 };
 
 const Kpis = ({
   isLoading,
   consumption,
+  period,
 }: Readonly<{
   isLoading: boolean;
   consumption: SavingsPlanFlavorConsumption;
+  period: string;
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-savings-plan/src/components/Dashboard/Kpis/Kpis.tsx:77]: I find it pretty confusing to have a period as a string

File: packages/manager/apps/pci-savings-plan/src/components/Dashboard/Kpis/Kpits.test.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest';
+import Kpis from './Kpis';
+import { useKpiData } from '@/hooks/useKpiData';
+import { SavingsPlanFlavorConsumption } from '@/types/savingsPlanConsumption.type';
+
+vi.mock('react-i18next', () => ({
+  useTranslation: () => ({
+    t: (key: string) => key,
+  }),
+}));
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-savings-plan/src/components/Dashboard/Kpis/Kpits.test.tsx:12]: I think you could mock it in the setupTests file instead

File: packages/manager/apps/pci-savings-plan/src/components/Dashboard/Kpis/Kpits.test.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      />,
+    );
+
+    expect(screen.getByText('dashboard_kpis_active_plans_name')).toBeTruthy();
+    expect(screen.getByText('dashboard_kpis_usage_percent_name')).toBeTruthy();
+    expect(
+      screen.getByText('dashboard_kpis_coverage_percent_name'),
+    ).toBeTruthy();
+
+    expect(screen.queryByText('dashboard_kpis_saved_amount_name')).toBeNull();
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-savings-plan/src/components/Dashboard/Kpis/Kpits.test.tsx:60]:
```suggestion
    expect(screen.queryByText('dashboard_kpis_saved_amount_name')).not.toBeInTheDocument();
```

File: packages/manager/apps/pci-savings-plan/src/components/Dashboard/Kpis/Kpits.test.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    const currentPeriod = 'janvier 2025';
+    render(
+      <Kpis
+        isLoading={false}
+        consumption={dummyConsumption}
+        period={currentPeriod}
+      />,
+    );
+
+    expect(screen.getByText('dashboard_kpis_active_plans_name')).toBeTruthy();
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-savings-plan/src/components/Dashboard/Kpis/Kpits.test.tsx:54]:
```suggestion
    expect(screen.getByText('dashboard_kpis_active_plans_name')).toBeInTheDocument();
```

File: packages/manager/apps/pci-savings-plan/src/hooks/useKpiData.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+    if (!consumption?.periods?.length) {
+      return {
+        computedUsagePercent: defaultMessage,
+        computedCoveragePercent: defaultMessage,
+        computedActivePlans: defaultMessage,
+        computedSavedAmount: defaultMessage,
+        computedWithoutSavedAmount: null,
+      };
+    }
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-savings-plan/src/hooks/useKpiData.tsx:55]: This should be at the top of the call since others values will be computed but not used

Also the test if consumption is null or undefined shouldn't be there since you declared it as not nullable

File: packages/manager/apps/pci-savings-plan/src/components/Dashboard/Kpis/Kpis.tsx [hunk 7]
```diff
@@ -44,58 +57,109 @@ const Kpi = ({
...
+        computedCoveragePercent: 0,
+        computedSavedAmount: 0,
+        computedWithoutSavedAmount: 0,
       };
     }
+
+    const { saved_amount: rawSavedAmount = 0, total_price: rawTotalPrice = 0 } =
+      consumption.fees ?? {};
+    const savedAmount = Number(rawSavedAmount);
+    const totalPrice = Number(rawTotalPrice);
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-savings-plan/src/components/Dashboard/Kpis/Kpis.tsx:114]: Why is that here ? Is it to cast ? If so I think it should be handled elsewhere

File: packages/manager/apps/pci-savings-plan/src/utils/kpi/utils.ts [hunk 8]
```diff
@@ -1,17 +1,87 @@
...
+export enum Period {
+  utilization = 'utilization',
+  coverage = 'coverage',
+}
+
 export const getPercentValue = (
   consumption: SavingsPlanFlavorConsumption,
-  key: 'utilization' | 'coverage',
-): string => {
+  key: Period.utilization | Period.coverage,
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-savings-plan/src/utils/kpi/utils.ts:10]:
```suggestion
  key: Period,
```

File: packages/manager/apps/pci-savings-plan/src/utils/kpi/utils.ts [hunk 9]
```diff
@@ -1,17 +1,87 @@
...
   }
 
   const average = values.reduce((sum, value) => sum + value, 0) / values.length;
-  return `${Math.floor(average)}%`;
+  return Math.floor(average);
+};
+
+export const isCurrentPeriod = (period: string): boolean => {
+  const now = new Date();
+  const periodDate = new Date(period);
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-savings-plan/src/utils/kpi/utils.ts:26]: I think cast should not be handled here

File: packages/manager/apps/pci-savings-plan/src/utils/kpi/utils.ts [hunk 10]
```diff
@@ -1,17 +1,87 @@
...
+    periodDate.getMonth() === now.getMonth()
+  );
+};
+
+export const calculateComputedPercents = (
+  consumption: SavingsPlanFlavorConsumption,
+  formatPrice: (
+    price: number,
+    options: { decimals: number; unit: number },
+  ) => string,
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-savings-plan/src/utils/kpi/utils.ts:43]: format should be handled by component

File: packages/manager/apps/pci-savings-plan/src/utils/kpi/utils.ts [hunk 11]
```diff
@@ -1,17 +1,87 @@
...
+) => {
+  const { fees } = consumption;
+  const { saved_amount: rawSavedAmount = 0, total_price: rawTotalPrice = 0 } =
+    fees ?? {};
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-savings-plan/src/utils/kpi/utils.ts:47]:
```suggestion
  const { saved_amount: rawSavedAmount = 0, total_price: rawTotalPrice = 0 } =
    consumption.fees ?? {};
```

File: packages/manager/apps/pci-savings-plan/src/utils/kpi/utils.ts [hunk 12]
```diff
@@ -1,17 +1,87 @@
...
+      computedSavedAmount: null,
+      computedWithoutSavedAmount: null,
+    };
+  }
+
+  const savedAmount = Number(rawSavedAmount);
+  const totalPrice = Number(rawTotalPrice);
+
+  const totalAmountWithoutSavingsPlans =
+    Math.min(savedAmount, totalPrice) <= 0 ? 0 : savedAmount + totalPrice;
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-savings-plan/src/utils/kpi/utils.ts:60]: Not sure to understand this condition, do you check for 0 value or something else ?

File: packages/manager/apps/pci-savings-plan/src/utils/kpi/utils.ts [hunk 13]
```diff
@@ -1,17 +1,87 @@
...
+  value: number | string,
+  titleTranslationKey: string,
+): string => {
+  if (!value) {
+    return 'dashboard_kpis_not_available';
+  }
+  const percentageKeys = [
+    'dashboard_kpis_usage_percent_name',
+    'dashboard_kpis_coverage_percent_name',
+  ];
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-savings-plan/src/utils/kpi/utils.ts:83]: Should be outside

File: packages/manager/apps/pci-savings-plan/src/utils/kpi/utils.ts [hunk 14]
```diff
@@ -1,17 +1,87 @@
...
+  return percentageKeys.includes(titleTranslationKey)
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-savings-plan/src/utils/kpi/utils.ts:84]: I don't understand what you are checking here. Also the titleTranslationKey isn't used for anything else

File: packages/manager/apps/pci-savings-plan/src/utils/test/setupTests.tsx [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+import '@testing-library/jest-dom';
+
+vi.mock('react-i18next', () => ({
+  useTranslation: () => ({
+    t: (key: string) => key,
+  }),
+  Trans: (props: any) => (
+    <span data-testid="trans">
+      {props.i18nKey} {JSON.stringify(props.values)}
+    </span>
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-savings-plan/src/utils/test/setupTests.tsx:12]: Prefer checking if the translation function has been called with specific parameters instead

**Review by SimonChaumet — APPROVED** [review]
